ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -1,223 +1,89 @@
-          type={NameType.ETHEREUM_ADDRESS}
-        />
-      </Box>
-    </>
-  );
-
-  return (
-    <StaticSimulation
-      title={t('simulationDetailsTitle')}
-      titleTooltip={t('simulationDetailsTitleTooltip')}
-      description={t('simulationDetailsApproveDesc')}
-      simulationHeading={
-        isNFT ? t('simulationApproveHeading') : t('spendingCap')
-      }
-      simulationElements={simulationElements}
-    />
-  );
-};
-
-const SpendingCapGroup = ({
-  tokenSymbol,
-  decimals,
-}: {
-  tokenSymbol: string;
-  decimals: string;
-}) => {
-  const t = useI18nContext();
-
-  const transactionMeta = useSelector(
-    currentConfirmationSelector,
-  ) as TransactionMeta;
-
-  const { tokenAmount, formattedTokenNum, value } = useApproveTokenSimulation(
-    transactionMeta,
-    decimals,
-  );
-
-  const SpendingCapElement = (
-    <ConfirmInfoRowText
-      text={
-        tokenAmount === UNLIMITED_MSG
-          ? `${t('unlimited')} ${tokenSymbol}`
-          : `${formattedTokenNum} ${tokenSymbol}`
-      }
-      onEditClick={
-        transactionMeta.type === TransactionType.tokenMethodIncreaseAllowance
-          ? () => console.log('TODO on a following ticket')
-          : undefined
-      }
-      editIconClassName="edit-spending-cap"
-    />
-  );
-
-  if (!value) {
-    return null;
-  }
-
-  return (
-    <>
-      <ConfirmInfoRowDivider />
-
-      <ConfirmInfoRow
-        label={t('spendingCap')}
-        tooltip={t('spendingCapTooltipDesc')}
-      >
-        {tokenAmount === UNLIMITED_MSG ? (
-          <Tooltip title={formattedTokenNum}>{SpendingCapElement}</Tooltip>
-        ) : (
-          SpendingCapElement
-        )}
-      </ConfirmInfoRow>
-    </>
-  );
-};
+  const [isOpenEditSpendingCapModal, setIsOpenEditSpendingCapModal] =
+    useState(false);
+  const [customSpendingCap, setCustomSpendingCap] = useState('');
 
-const SpendingCap = () => {
-  const t = useI18nContext();
-
-  const transactionMeta = useSelector(
-    currentConfirmationSelector,
-  ) as TransactionMeta;
-
-  const { userBalance, tokenSymbol, decimals } = useAssetDetails(
-    transactionMeta.txParams.to,
-    transactionMeta.txParams.from,
-    transactionMeta.txParams.data,
+  const { decimals } = useAssetDetails(
+    transactionMeta?.txParams?.to,
+    transactionMeta?.txParams?.from,
+    transactionMeta?.txParams?.data,
   );
 
-  const accountBalance = new BigNumber(userBalance || '0')
-    .dividedBy(new BigNumber(10).pow(Number(decimals || '0')))
-    .toNumber();
-
-  const { pending } = useApproveTokenSimulation(
+  const customTxParamsData = useMemo(() => {
+    return getCustomTxParamsData(transactionMeta?.txParams?.data, {
+      customPermissionAmount: customSpendingCap || '0',
+      decimals,
+    });
+  }, [customSpendingCap, transactionMeta?.txParams?.data, decimals]);
+
+  const { value: estimatedGasLimit } = useAsyncResult(async () => {
+    return await estimateGas({
```

Comment:
Gas is a scary one that got really complex in the legacy confirmations, could we encapsulate this behind a `useTransactionGasEstimate` so there's a single definition of how we calculate the gas for the current transaction?